Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableSystemComponent.cpp

```diff
@@ -144,6 +170,92 @@ namespace AzFramework
         AZ_TracePrintf("Spawnables", "Generation %i of the root spawnable has been released.\n", generation);
     }
 
+    EntitySpawnTicket SpawnableSystemComponent::CreateSpawnTicket(const SpawnableAssetRef& spawnableAsset)
+    {
+        return EntitySpawnTicket(spawnableAsset.m_asset);
+    }
+
+    bool SpawnableSystemComponent::Spawn(
+        EntitySpawnTicket spawnTicket,
+        AZ::EntityId parentId,
+        AZ::Vector3 translation,
+        AZ::Vector3 rotation,
+        float scale)
+    {
+        if (!spawnTicket.IsValid())
+        {
+            AZ_Error("Spawnables", false, "EntitySpawnTicket used for spawning is invalid.");
+            return false;
+        }
+
+        auto preSpawnCB = [this, parentId, translation, rotation, scale](
+                              [[maybe_unused]] EntitySpawnTicket::Id ticketId,
+            SpawnableEntityContainerView view)
+        {
+            AZStd::lock_guard lock(m_mutex);
+
+            AZ::Entity* rootEntity = *view.begin();
+            TransformComponent* entityTransform = rootEntity->FindComponent<TransformComponent>();
+
+            if (entityTransform)
+            {
+                AZ::Vector3 rotationCopy = rotation;
+                AZ::Quaternion rotationQuat = AZ::Quaternion::CreateFromEulerAnglesDegrees(rotationCopy);
+
+                TransformComponentConfiguration transformConfig;
+                transformConfig.m_parentId = parentId;
+                transformConfig.m_localTransform = AZ::Transform(translation, rotationQuat, scale);
+                entityTransform->SetConfiguration(transformConfig);
+            }
+        };
+
+        auto spawnCompleteCB =
+            [this,
+             spawnTicket]([[maybe_unused]] EntitySpawnTicket::Id ticketId, SpawnableConstEntityContainerView view)
+        {
+            AZStd::lock_guard lock(m_mutex);
+
+            SpawnableResult spawnableResult;
+            // SpawnTicket instance is cached instead of SpawnTicketId to simplify managing its lifecycle on Script Canvas
+            // and to provide easier access to it in OnSpawnCompleted callback
+            spawnableResult.m_spawnTicket = spawnTicket;
+            spawnableResult.m_entityList.reserve(view.size());
+            for (const AZ::Entity* entity : view)
+            {
+                spawnableResult.m_entityList.emplace_back(entity->GetId());
+            }
```

Comment:
while I agree, there's a case when this can happen, this seems more of a user error to me. E.g. someone has a user component that destroys the entity on activate.